### ISSUE TITLE
[Feature] Kanban Board

### ISSUE THREAD

**Issue by brendanperry** [cryptee/web-client#66]
I’ve been using Cryptee a lot for day to day activities and to-do type workloads. I enjoy using a kanban board for organizing tasks. I would like to stop using Trello because their privacy model is not great. I think this would be a great addition.

Thanks for your time, and great job with all the hard work that has been put in thus far!

**Comment by johnozbay** [comment]
Hey Brendan! 👋🏻
Thanks a lot for your kind words, and really happy to hear you're enjoying Cryptee! 

Kanban Boards have been on our list of things to do for a while now! We're big fans too.
Right now we're working on a major v2.0 update on the whole platform. Be it design / features / speed etc all will improve. It's a big revamp of everything. 

Once this is out of the way we will start looking into things like Kanban / other productivity tools & integrate more workflows into the suite! 

I'll keep this thread open, and collect more feedback from our community, feel free to add in all the features you'd like to see, and we'll make it happen! 

All the very best,
J

**Comment by johnozbay** [comment]
Hey Brendan, 

Thanks a lot for your patience with this! 

Due to the overall complexity of kanban boards (or to put it differently – how this will either over-complicate Cryptee Docs, or add an entirely new product to our services line-up), in the interest of making our existing apps better, and making the best use of our limited resources, we won't be tackling this!

One angle we considered primarily while building v3, was to build different 'editors' into Cryptee Docs. So if you open a `.css` file for example, it would open a code editor, or if you open a Cryptee Doc, it would open our regular editor etc... So if there existed a Trello file format for kanban boards (let's call this `.trel` for the sake of example) we could theoretically make a 'kanban' editor, and that way Cryptee Docs could open / edit kanban boards. But with v3, we realized we should simplify things as much as we can, and focus heavily on usability and functionality. – so we decided not to take this approach either.

If you've got ideas about how we can somehow build this into the platform without either making it a separate app OR how we can build it into Cryptee Docs easily, let us know. We'll be open to ideas / would love to hear how else you'd like to use Cryptee! 

Many thanks for your patience, and for filing this! 🙏🏻

Best,
J

**Comment by brendanperry** [comment]
I saw that I had asked for this a few weeks back and had totally forgot. I do agree with you that this is a bit off the board. 

My original goal was less about Kanban boards and really more about being able to properly handle a todo list within Cryptee. However, given there still isn't a way for specifically timed notifications, adding this would not be enough anyways. 

I think it's a great idea to make Cryptee better at what it's meant for because that's why I've stuck with it anyways. (Also hoping for spreadsheets soon🙂 lol)

**Comment by johnozbay** [comment]
Thanks for understanding! 

Yeah notifications are the tricky part. Apps themselves can handle them (even on iOS there are a few tricks we can do for this to add notifications in our PWA) – but if the server can't see your documents, nor its contents, nor the time notifications should be sent out, then it can't trigger notifications etc, making this part pretty much impossible.  

Good news is that spreadsheets are still on the table. Given that there's very clear and well established formats for them like (xls / csv etc) we're looking into how we can build our editor to work with them as expected. 

tldr; if we add spreadsheets, everyone will rightfully expect things like formulas / copy paste from excel etc to work – but formulas can contain anything from wildly long math formulas all the way to malicious scripts [something which microsoft has been fighting for years now... And we can't support 'some' formulas, we'd have to support all or none to not break compatibility while importing / exporting etc. + there's the legal liability. imagine if you import your company's budget sheets, and there's a math error, so you end up making a $1M mistake. who is to blame here etc? so it's a rather tricky thing to tackle, and we're still looking into it  🤓
